### PR TITLE
Extend TensorComputeOp to allow scalar inputs (#2606).

### PR DESCRIPTION
**Pull request by jdavies-huawei**

### CODE REVIEW

**Comment by jdavies-huawei** [comment]
@xqdan @ZihengJiang @tmoreau89 please review.

File: include/tvm/tensor_intrin.h [hunk 1]
```diff
@@ -49,6 +49,11 @@ class TensorIntrinNode : public Node {
    *  When it is a constant, it means we can only take data in that shape.
    */
   Array<Buffer> buffers;
+  /*! \brief List of scalar variables, used in body. These placeholders
+   *  will be bound to expressions passed in when the TensorIntrin is called
+   * from a TensorComputeOp.
+   */
+  Array<Var> scalar_params;
```
Comment (ZihengJiang) [inline include/tvm/tensor_intrin.h]: `scalar_inputs`, to be consistent with class `Operation`

**Comment by icemelon9** [comment]
Why don't we use 0-dim tensor to represent scalar input?

**Comment by junrushao1994** [comment]
> Why don't we use 0-dim tensor to represent scalar input?

Me wondering as well

**Pull request closed without merging**